loader/loader.go

```diff
@@ -285,25 +285,27 @@ func Load(configDetails types.ConfigDetails, options ...func(*Options)) (*types.
 	return LoadWithContext(context.Background(), configDetails, options...)
 }
 
-// LoadWithContext reads a ConfigDetails and returns a fully loaded configuration
+// LoadWithContext reads a ConfigDetails and returns a fully loaded configuration as a compose-go Project
 func LoadWithContext(ctx context.Context, configDetails types.ConfigDetails, options ...func(*Options)) (*types.Project, error) {
-	if len(configDetails.ConfigFiles) < 1 {
-		return nil, errors.New("No files specified")
+	opts := toOptions(configDetails, options)
+	dict, err := loadModelWithContext(ctx, configDetails, opts)
+	if err != nil {
+		return nil, err
 	}
+	return modelToProject(dict, opts, configDetails)
+}
 
-	opts := &Options{
-		Interpolate: &interp.Options{
-			Substitute:      template.Substitute,
-			LookupValue:     configDetails.LookupEnv,
-			TypeCastMapping: interpolateTypeCastMapping,
-		},
-		ResolvePaths: true,
-	}
+// LoadModelWithContext reads a ConfigDetails and returns a fully loaded configuration as a yaml dictionary
+func LoadModelWithContext(ctx context.Context, configDetails types.ConfigDetails, options ...func(*Options)) (map[string]any, error) {
+	opts := toOptions(configDetails, options)
+	return loadModelWithContext(ctx, configDetails, opts)
+}
 
-	for _, op := range options {
-		op(opts)
+// LoadModelWithContext reads a ConfigDetails and returns a fully loaded configuration as a yaml dictionary
+func loadModelWithContext(ctx context.Context, configDetails types.ConfigDetails, opts *Options) (map[string]any, error) {
```

Comment:
```suggestion
// loadModelWithContext reads a ConfigDetails and returns a fully loaded configuration as a yaml dictionary
func loadModelWithContext(ctx context.Context, configDetails types.ConfigDetails, opts *Options) (map[string]any, error) {
```

Reply:
I made this one intentionally public, as `LoadProjectWithContext` for third party directly passing configDetails